benchmarks/benchmark_fp6.py

```diff
@@ -7,7 +7,7 @@
 from tqdm import tqdm
 
 
-def benchmark(m: int, k: int, n: int):
+def benchmark(m: int, n: int, k: int):
```

Comment:
is this change intentional? usually we talk about matmuls as mkn i.e. m x k activation and k x n weight (odd to reverse them and i'm unsure if benchmarks previously were assuming the other ordering)

Reply:
Looking at this again, I indeed made a mistake. The benchmark results are still correct, it's just the list of shapes are different.

I took the benchmark shapes from here: https://github.com/usyd-fsalab/fp6_llm/blob/ce76774bcfc26b325c1b558abcf1935026d9abbc/tests/python/run.sh - It's a bit confusing since the author use different variable names...

The code generating the list of shapes (under `__name__ == "__main__"`) are correct (follow the author), and it calls `benchmark(m, n, k)`. If you think we should benchmark a different sets of shapes, it should be good too!

In summary, this change corrects my previous mistake.

Reply:
Yes, it's intentional. The function signature was `def benchmark(m: int, k: int, n: int):`, but arguments were passed as `(m, n, k)`, so I thought that that was unnecessarily confusing and wanted to change the ordering in either the function call or the function signature. In the function itself, the shapes become m x k for the activation and n x k for the weight. 

I see one benchmark example (`benchmark_gpu_sparsity`, see below) where the ordering is m, k, n, so let me change the function signature back to that ordering for consistency.
https://github.com/pytorch/ao/blob/4b5b5ee599f47eb9245c370caa559641a1f0fbb2/benchmarks/benchmark_gpu_sparsity.py#L25

Reply:
Missed your comment before I posted my own @gau-nernst. Thanks for clarifying! I actually noticed that `m` gets passed as `n` to the actual kernel, which is slightly confusing. If you don't mind, I'll change this for consistency. I don't think it should affect the results, expect that `m` will be switched by `n` in the performance table.

Reply:
@tobiasvanderwerff You mean `k` and `n` right? Your current change looks correct. Yea it doesn't affect the results, it will only show different shapes instead.

Reply:
What I meant is slightly different. I'm referring to the fact that the original authors do some odd switching of the shapes in `fp6_linear.cu`. The arguments that get passed are `_in_feats` (activations of shape m x k) and `_weights` (shape n x k), but then they unpack the shapes as `M = _weights.size(0)`, `K = _in_feats.shape(0)`, `N = _in_feats.shape(1)` (see below). 

https://github.com/pytorch/ao/blob/4b5b5ee599f47eb9245c370caa559641a1f0fbb2/torchao/csrc/cuda/fp6_llm/fp6_linear.cu#L150-L158

So even though we pass the arguments correctly to the benchmark function as m, k, n, the names get switched inside the kernel. Anyway, this is mainly confusing when debugging the kernel, but it might actually be fine to just leave it as is, since it does not affect the shapes/results that get passed back from the kernel.